hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeysRenameRequest.java

```diff
@@ -0,0 +1,315 @@
+import static org.apache.hadoop.ozone.OzoneConsts.RENAMED_KEYS_MAP;
+import static org.apache.hadoop.ozone.OzoneConsts.UNRENAMED_KEYS_MAP;
+import static org.apache.hadoop.ozone.OzoneConsts.VOLUME;
+import static org.apache.hadoop.ozone.om.lock.OzoneManagerLock.Resource.BUCKET_LOCK;
+
+/**
+ * Handles rename keys request.
+ */
+public class OMKeysRenameRequest extends OMKeyRequest {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(OMKeysRenameRequest.class);
+
+  public OMKeysRenameRequest(OMRequest omRequest) {
+    super(omRequest);
+  }
+
+  @Override
+  public OMRequest preExecute(OzoneManager ozoneManager) throws IOException {
+
+    RenameKeysRequest renameKeys = getOmRequest().getRenameKeysRequest();
+    Preconditions.checkNotNull(renameKeys);
+
+    List<RenameKeyRequest> renameKeyList = new ArrayList<>();
+    for (RenameKeyRequest renameKey : renameKeys.getRenameKeyRequestList()) {
+      // Set modification time.
+      KeyArgs.Builder newKeyArgs = renameKey.getKeyArgs().toBuilder()
+          .setModificationTime(Time.now());
+      renameKey.toBuilder().setKeyArgs(newKeyArgs);
+      renameKeyList.add(renameKey);
+    }
+    RenameKeysRequest renameKeysRequest = RenameKeysRequest
+        .newBuilder().addAllRenameKeyRequest(renameKeyList).build();
+    return getOmRequest().toBuilder().setRenameKeysRequest(renameKeysRequest)
+        .setUserInfo(getUserInfo()).build();
+  }
+
+  @Override
+  @SuppressWarnings("methodlength")
+  public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
+      long trxnLogIndex, OzoneManagerDoubleBufferHelper omDoubleBufferHelper) {
+
+    RenameKeysRequest renameKeysRequest = getOmRequest().getRenameKeysRequest();
+    OMClientResponse omClientResponse = null;
+    // fromKeyName -> toKeyNmae
+    List<RenameKeyArgs> unRenamedKeys = new ArrayList<>();
+
+    List<OmRenameKeyInfo> renameKeyInfoList = new ArrayList<>();
+
+    OMMetrics omMetrics = ozoneManager.getMetrics();
+    omMetrics.incNumKeyRenames();
+
+    AuditLogger auditLogger = ozoneManager.getAuditLogger();
+
+
+    OMResponse.Builder omResponse = OmResponseUtil.getOMResponseBuilder(
+        getOmRequest());
+
+    OMMetadataManager omMetadataManager = ozoneManager.getMetadataManager();
+    IOException exception = null;
+    OmKeyInfo fromKeyValue = null;
+
+    Result result = null;
+    Map<String, String> auditMap = null;
+    RenameKeyRequest renameRequest = null;
+    String volumeName = null;
+    String bucketName = null;
+    String fromKeyName = null;
+    String toKeyName = null;
+    boolean acquiredLock = false;
+    boolean renameStatus = true;
+    try {
+      for (RenameKeyRequest renameKeyRequest : renameKeysRequest
+          .getRenameKeyRequestList()) {
+        OzoneManagerProtocolProtos.KeyArgs keyArgs =
+            renameKeyRequest.getKeyArgs();
+
+        volumeName = keyArgs.getVolumeName();
+        bucketName = keyArgs.getBucketName();
```

Comment:
Please add support for bucket links.  Example:

https://github.com/apache/hadoop-ozone/blob/3571d7e72688b008c2667997cb4d824ccc9a8a3e/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyRenameRequest.java#L134-L136

`auditMap` needs to be created beforehand, eg.:

https://github.com/apache/hadoop-ozone/blob/3571d7e72688b008c2667997cb4d824ccc9a8a3e/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyRenameRequest.java#L106